models.py

```diff
@@ -0,0 +1,266 @@
+"""SQLAlchemy models for Invoice App."""
+
+from datetime import datetime
+
+from flask_bcrypt import Bcrypt
+from flask_sqlalchemy import SQLAlchemy
+from datetime import datetime
+from sqlalchemy.ext.declarative import declarative_base
+
+
+
+Base = declarative_base()
+
+bcrypt = Bcrypt()
+db = SQLAlchemy()
+
+def connect_db(app):
+    """Connect this database to provided Flask app."""
+
+    db.app = app
+    db.init_app(app)
+
+
+# class BaseModel(Base):
+#     __abstract__ = True
+#     created_date = db.Column(db.DateTime, default=datetime.utcnow)
+#     updated_date = db.Column(db.DateTime, default=datetime.utcnow, onupdate=datetime.utcnow)
+#     deleted_date = db.Column(db.DateTime)
+
+
+class User(db.Model):
+    """User in the system."""
+
+    __tablename__ = 'users'
+
+    id = db.Column(
+        db.Integer,
+        primary_key=True,
+    )
+    email = db.Column(
+        db.Text,
+        nullable=False,
+        unique=True,
+    )
+    username = db.Column(
+        db.Text,
+        nullable=False,
+        unique=True,
+    )
+    password = db.Column(
+        db.Text,
+        nullable=False,
+    )
+
+    def __repr__(self):
+        return f"<User #{self.id}: {self.username}, {self.email}>"
+
+    @classmethod
+    def signup(cls, username, email, password, image_url):
+        """Sign up user.
+
+        Hashes password and adds user to system.
+        """
+
+        hashed_pwd = bcrypt.generate_password_hash(password).decode('UTF-8')
+
+        user = User(
+            username=username,
+            email=email,
+            password=hashed_pwd,
+            image_url=image_url,
+        )
+
+        db.session.add(user)
+        return user
+
+    @classmethod
+    def authenticate(cls, username, password):
```

Comment:
Let's move these methods into a class called UserService. This service is responsible for interacting with your data models.